plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/button/frontend.ts

```diff
@@ -51,14 +55,58 @@ const { state: addToCartWithOptionsState } = store< AddToCartWithOptionsStore >(
 	{ lock: universalLock }
 );
 
+const isCartItemMatched = (
+	cartItem: OptimisticCartItem,
+	selectedItem: CartVariationItem[]
+) => {
+	if (
+		! Array.isArray( cartItem.variation ) ||
+		! Array.isArray( selectedItem )
+	)
+		return false;
+
+	// Incase the attributes list length is different in both the objects
+	if ( cartItem.variation.length !== selectedItem.length ) {
+		return false;
+	}
+
+	return cartItem.variation.every(
+		( {
+			raw_attribute,
+			value,
+		}: {
+			raw_attribute: string;
+			value: string;
+		} ) =>
+			selectedItem.some( ( item ) => {
+				return (
+					item.attribute === raw_attribute &&
+					( item.value.toLowerCase() === value.toLowerCase() ||
+						( item.value && value === '' ) ) // Handle "any" attribute type
+				);
+			} )
+	);
+};
+
 const productButtonStore = {
 	state: {
 		get quantity(): number {
-			const product = wooState.cart?.items.find(
+			const products = wooState.cart?.items.filter(
 				( item ) => item.id === state.productId
 			);
 
-			return product?.quantity || 0;
+			// Return the product quantity when the item is a non-variable product.
+			if ( products[ 0 ]?.type !== 'variation' ) {
+				return products[ 0 ]?.quantity;
```

Comment:
Just in case, should this still be:

```suggestion
				return products[ 0 ]?.quantity || 0;
```